### PR TITLE
Usman Zaver - Weatherman

### PR DESCRIPTION
**Pull request by usmanzaver**

### CODE REVIEW

File: weatherman/calculations_file.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+    for filename in os.listdir(path):
+
+        if year in filename:
+            files_list.append(filename)
+            total_files = total_files + 1
+    return files_list, total_files
+
+
+# Validating number of arguments
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

File: weatherman/calculations_file.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+def no_of_arguments():
+    total_arguments = len(sys.argv)
+    print("arguments : ", total_arguments)
+    if total_arguments > 3 and total_arguments % 2 == 0:
+        return total_arguments
+    else:
+        print("invalid arguments")
+        return False
+
+# Map number to month
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

File: weatherman/calculations_file.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    else:
+        if month.isalpha() is False:
+            month = int(month)
+            month = datetime.datetime(year, month, 2)
+            return(month.strftime("%b"))
+
+        else:
+            return month
+
+# no_of Records in a file
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

File: weatherman/calculations_file.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    for i in range (len(arguments_list)):
+        #print(arguments_list[i][j])
+        year, monthz = arguments_list[i].split("/")
+        monthz = map_month(int(year), monthz)
+        print(path, year, monthz, argument)
+        main(path, year, monthz, argument)
+
+def main(path, year, month, arg):
+
+    # Getting the files_list and the total files required
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    files_list, total_files = my_files(path, year, month)
+    # Create objects according to the number of files
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+    obj_weather = create_weather_objects(total_files)
+
+    # Filling objects
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+        path, total_files, files_list, obj_weather)
+
+    # Creating a list of Parse Class objects
+    obj_parser = create_parsing_objects(total_files)
+
+    # Populate parser objects
+    obj_parser = populate_parser_objects(
+        obj_parser, total_files, obj_weather)
+
+    # populating the datastructure
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+    obj_weather = create_weather_objects(total_files)
+
+    # Filling objects
+    obj_weather = populate_weather_objects(
+        path, total_files, files_list, obj_weather)
+
+    # Creating a list of Parse Class objects
+    obj_parser = create_parsing_objects(total_files)
+
+    # Populate parser objects
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+    obj_parser = populate_parser_objects(
+        obj_parser, total_files, obj_weather)
+
+    # populating the datastructure
+    for i in range(total_files):
+        obj_parser[i].parse(obj_weather[i])
+
+    # Creating Calculations object
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+    obj_calculations = calculations_file.Calculations()
+
+    # Creating object which holds calculations
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+    calculated_data = data_file.Calculations_results
+
+    # Populating the calculations datastructure
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/calculations_file.py [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+        obj_parser)
+
+    calculated_data.mean_humid = obj_calculations.calculate_avg_mean_humid(
+        obj_parser[0])
+    calculated_data.max_avg_temp = obj_calculations.calculate_avg_max_temp(
+        obj_parser[0])
+    calculated_data.min_avg_temp = obj_calculations.calculate_avg_min_temp(
+        obj_parser[0])
+
+    # Creating Report object
```
Comment (mumer92) [inline weatherman/calculations_file.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

no need for seperate comments, your variable names should be self identifying

File: weatherman/weatherman.py [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+    else:
+        year = year + "_" + month
+    for filename in os.listdir(path):
+        if year in filename:
+            files_list.append(filename)
+            total_files = total_files + 1
+    return files_list, total_files
+
+
+# Validating number of arguments
```
Comment (mumer92) [inline weatherman/weatherman.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

File: weatherman/weatherman.py [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+def no_of_arguments():
+    total_arguments = len(sys.argv)
+    print("arguments : ", total_arguments)
+    if total_arguments > 3 and total_arguments % 2 == 0:
+        return total_arguments
+    else:
+        print("invalid arguments")
+        return False
+
+# Map number to month
```
Comment (mumer92) [inline weatherman/weatherman.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

File: weatherman/weatherman.py [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+        return month
+    else:
+        if month.isalpha() is False:
+            month = int(month)
+            month = datetime.datetime(year, month, 2)
+            return(month.strftime("%b"))
+        else:
+            return month
+
+# no_of Records in a file
```
Comment (mumer92) [inline weatherman/weatherman.py]: use `docstring` for comments'
[https://www.python.org/dev/peps/pep-0257/](url)

File: weatherman/weatherman.py [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+                print("+", end='')
+        sys.stdout.write("\033[0;0m")
+
+        if is_min_int and is_max_int is True:
+            print(" ", str(min_temp) + "C", "-", str(max_temp) + "C")
+        else:
+            print("NULL")
+
+
+def main(path, year, month, arg):
```
Comment (mumer92) [inline weatherman/weatherman.py]: break down `main` into smaller functions where you are using comments, each function should perform its task and return vale, your `main` function will not be cluttered.

File: weatherman/calculations_file.py [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+
+
+def create_parsing_objects(total_files):
+    obj_parser = list()
+    for i in range(total_files):
+        obj_parser.append(data_file.Parser())
+    return obj_parser
+
+
+def populate_parser_objects(obj_parser, total_files, obj_weather):
```
Comment (mumer92) [inline weatherman/calculations_file.py]: instead or `populate_parser_objects`, you can create class `Temperature` to hold array objects, that way you can avoid passing indices to obtain value.

File: weatherman/weatherman.py (hunk 16 above)
Comment (usmanzaver) [reply weatherman/weatherman.py]: I've tried reducing the main function.

File: weatherman/calculations_file.py (hunk 17 above)
Comment (usmanzaver) [reply weatherman/calculations_file.py]: I've completely removed the parser object as it looked redundant.

**Pull request merged**